Add data science notebook image

atheo89 opened the pull request:
Added data science notebook image

## Description
Files that have been added:
- Dockerfile
- Kustomize objects
- requirment.in 

Files that have been modified:
- Makefile


## How Has This Been Tested?
`podman run -it --rm -p 8888:8888 quay.io/{USER_REGISTRY}:jupyter-data-science-ubi8-python-3.8`

## Merge criteria:
<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

samuelvl commented:
/test all

samuelvl commented on jupyter/datascience/ubi8-python-3.8/Dockerfile:
```diff
@@ -0,0 +1,21 @@
+
+COPY requirements.txt ./
+
+# Fix permissions to support pip in Openshift environments
+RUN chmod -R g+w /opt/app-root/lib/python3.8/site-packages && \
+      fix-permissions /opt/app-root -P
```
This should happen after `pip install`

samuelvl commented on jupyter/datascience/ubi8-python-3.8/requirements.in:
```diff
@@ -0,0 +1,18 @@
+boto3==1.17.11
+scipy==1.6.2
+
+# JupyterLab extensions
+jupyter-resource-usage==0.6.1
+jupyterlab-s3-browser==0.10.1
+jupyterlab-widgets==3.0.3
+jupyterlab-lsp==3.10.0
+jupyterlab-git==0.30.0
+jupyter-bokeh==3.0.4
+plotly==5.5.0
+nbdime==3.0.0
```
`jupyterlab-git` and `nbdime` are already included in the minimal notebook image, no need to include them twice

samuelvl commented on jupyter/datascience/ubi8-python-3.8/requirements.in:
```diff
@@ -0,0 +1,18 @@
+boto3==1.17.11
+scipy==1.6.2
+
+# JupyterLab extensions
```
Some extensions from the ODS CI framework are missing. See https://github.com/red-hat-data-services/ods-ci/blob/master/tests/Tests/500__jupyterhub/plugin-verification.robot#L17

samuelvl commented on Makefile:
```diff
@@ -67,6 +67,14 @@ jupyter-minimal-ubi8-python-3.8: base-ubi8-python-3.8
 
+# Build and push jupyter-datascience-ubi8-python-3.8 image to the registry
+.PHONY: jupyter-datascience-ubi8-python-3.8
+jupyter-datascience-ubi8-python-3.8: jupyter-minimal-ubi8-python-3.8
```
Documentation in README.md is missing for this new notebook

atheo89 commented:
Thanks for the review! I rebased my local repo to apply your comments. About the extensions from the ODS CI framework, the `jupyterlab_requirments` is missing on the old minimal image. I have added it now.

samuelvl commented on Makefile:
```diff
@@ -67,6 +67,14 @@ jupyter-minimal-ubi8-python-3.8: base-ubi8-python-3.8
+.PHONY: jupyter-datascience-ubi8-python-3.8
+jupyter-datascience-ubi8-python-3.8: jupyter-minimal-ubi8-python-3.8
+	$(call pip_compile,bootstrap/ubi8-python-3.8,\
+		base/ubi8-python-3.8/requirements.in jupyter/datascience/ubi8-python-3.8/requirements.in,\
```
We have to include `jupyter/minimal/ubi8-python-3.8/requirements.in` too.

samuelvl commented:
@atheo89 

> About the extensions from the ODS CI framework, the jupyterlab_requirments is missing on the old minimal image. I have added it now.

What do you mean? I can't find the `jupyterlab_requirements` package in the old minimal notebook:

```shell
$ podman run --rm quay.io/modh/odh-minimal-notebook-container:v1.17.0-9 pip show jupyterlab-requirements
WARNING: Package(s) not found: jupyterlab-requirements

$ podman run --rm quay.io/modh/odh-minimal-notebook-container:v1.17.0-9 pip show jupyterlab_requirements
WARNING: Package(s) not found: jupyterlab_requirements
```

atheo89 commented:
Sorry, I wasn't clear before, I added the `jupyterlab_requirements` into the [requirements.in ](https://github.com/atheo89/notebooks/blob/02586a49f3ca3217754db22638b7359d6c23257e/jupyter/datascience/ubi8-python-3.8/requirements.in#L11) in the new data science notebook image because it exists into [ods-ci](https://github.com/red-hat-data-services/ods-ci/blob/master/tests/Tests/500__jupyterhub/plugin-verification.robot#L17) Should I leave it or exclude it?

samuelvl commented on README.md:
````diff
@@ -51,6 +52,24 @@ You can use a different registry by overwriting the `IMAGE_REGISTRY` variable:
+Build the `jupyter-datascience-ubi8-python-3.8` by running the following command:
+
+```shell
+make jjupyter-datascience-ubi8-python-3.8
````
Small typo `s/jjupyter/jupyter`
